src/qmapshack/canvas/CCanvas.cpp

```diff
@@ -1119,9 +1119,20 @@ void CCanvas::displayInfo(const QPoint& px)
 
 poi_t CCanvas::findPOICloseBy(const QPoint& px) const
 {
+    poi_t poiItem = poi->findPOICloseBy(px);
```

Comment:
This is blocking the main thread operating the GUI. Probably it gets stuck in the mutex of the POI thread. This is no good.